udata/tests/__init__.py

```diff
@@ -1,260 +1,92 @@
+        return self.client.delete(url, data=data, **kwargs)
 
-class WebTestMixin(object):
-    user = None
+    def assertRedirects(self, response, location, message=None):
+        """
+        Checks if response is an HTTP redirect to the
+        given location.
+        :param response: Flask response
+        :param location: relative URL path to SERVER_NAME or an absolute URL
+        """
+        __tracebackhide__ = True
+        helpers.assert_redirects(response, location, message=message)
 
-    def _build_url(self, url, kwargs):
-        if 'qs' not in kwargs:
-            return url
-        qs = kwargs.pop('qs')
-        return '?'.join([url, url_encode(qs)])
+    def assertStatus(self, response, status_code, message=None):
+        __tracebackhide__ = True
+        helpers.assert_status(response, status_code, message=message)
 
     def full_url(self, *args, **kwargs):
-        with self.app.test_request_context(''):
-            return urljoin(request.url_root, url_for(*args, **kwargs))
-
-    def get(self, url, client=None, **kwargs):
-        url = self._build_url(url, kwargs)
-        return (client or self.client).get(url, **kwargs)
-
-    def post(self, url, data=None, client=None, **kwargs):
-        url = self._build_url(url, kwargs)
-        return (client or self.client).post(url, data=data, **kwargs)
-
-    def put(self, url, data=None, client=None, **kwargs):
-        url = self._build_url(url, kwargs)
-        return (client or self.client).put(url, data=data, **kwargs)
+        __tracebackhide__ = True
+        return helpers.full_url(*args, **kwargs)
 
-    def delete(self, url, data=None, client=None, **kwargs):
-        url = self._build_url(url, kwargs)
-        return (client or self.client).delete(url, data=data, **kwargs)
+    def login(self, user=None):
+        self.user = self.client.login(user)
+        return self.user
 
-    def assert_flashes(self, expected_message, expected_category='message'):
-        with self.client.session_transaction() as session:
-            try:
-                category, message = session['_flashes'][0]
-            except KeyError:
-                raise AssertionError('nothing flashed')
-            self.assertIn(expected_message, message)
-            self.assertEqual(expected_category, category)
 
-    def assert_not_flash(self):
-        with self.client.session_transaction() as session:
-            flashes = session.get('_flashes', [])
-            self.assertNotIn(
-                '_flashes', session,
-                'There is {0} unexpected flashed message(s): {1}'.format(
-                    len(flashes),
-                    ', '.join('"{0} ({1})"'.format(msg, cat)
-                              for cat, msg in flashes)
-                ))
+for code in 200, 201, 204, 400, 401, 403, 404, 410, 500:
+    name = 'assert{0}'.format(code)
+    helper = getattr(helpers, name)
+    setattr(WebTestMixin, name, lambda s, r, h=helper: h(r))
```

Comment:
👌

Reply:
A good developer is a lazy one :stuck_out_tongue_winking_eye: